File: programmers/완주하지못한선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map <string, int> p;
    for (string name : participant) {
        p[name]++;
    }
    for (string com : completion) {
        p[com]--;
    }
    for (auto a : p) {
        if (a.second > 0) {
            answer = a.first;
            break;
        }
    }
    return answer;
}

File: programmers/완주하지못한선수.py
participant = ["leo", "kiki", "eden"]
completion = ["eden", "kiki"]
import collections
# def solution(participant, completion):
#     answer = ''
#     participants = collections.Counter(participant)
#     for c in completion:
#         if c in participants:
#             participants[c] -= 1
#     print(participants)
#     for p in participants:
#         if participants[p] == 1:
#             answer = p
#             break
#     return answer
# print(solution(participant, completion))

def solution(participant, completion):
    answer = ''
    running = collections.Counter(participant) - collections.Counter(completion)
    return list((running.keys()))[0]

print(solution(participant, completion))